functions/selfReportToIWF.ts

```diff
@@ -0,0 +1,63 @@
+import { Context, ServerlessCallback } from '@twilio-labs/serverless-runtime-types/types';
+import axios from 'axios';
+import { v4 } from 'uuid';
+import {
+  bindResolve,
+  error400,
+  error500,
+  responseWithCors,
+  send,
+  functionValidator as TokenValidator,
+} from '@tech-matters/serverless-helpers';
+
+type EnvVars = {
+  IWF_API_CASE_URL: string;
+  IWF_REPORT_URL: string;
+  IWF_SECRET_KEY: string;
+};
+
+export type IWFSelfReportPayload = {
+  secret_key: string;
+  case_number: string;
+  user_age_range: '<13' | '13-15' | '16-17';
+};
+
+export type Body = {
+  user_age_range?: '<13' | '13-15' | '16-17';
+  request: { cookies: {}; headers: {} };
+};
+
+export const handler = TokenValidator(
+  async (context: Context<EnvVars>, event: Body, callback: ServerlessCallback) => {
+    const response = responseWithCors();
+    const resolve = bindResolve(callback)(response);
+
+    try {
+      const { user_age_range } = event;
+      if (!user_age_range) return resolve(error400('user_age_range'));
+
+      const body: IWFSelfReportPayload = {
+        secret_key: context.IWF_SECRET_KEY,
+        case_number: v4(),
```

Comment:
Do we need to record this anywhere on our end? @GPaoloni

Reply:
Yes, I think we want to use the one generated in HRM for this. @acedeywin you can check that Flex, before calling this function [here](https://github.com/techmatters/flex-plugins/blob/0e2c8109a53462d6fcb912a4f4f5c4a263904337/plugin-hrm-form/src/components/CSAMReport/CSAMReport.tsx#L137), already have access to a csam record, which [contains a UUID generated under `.csamReportId` property](https://github.com/techmatters/hrm/blob/810e8732f6858058f6529d1f04ddbe1b99666ac0/hrm-service/src/csam-report/csam-report.ts#L22).